internal/backends/sqlite/collection.go

```diff
@@ -386,6 +387,107 @@ func (c *collection) CreateIndexes(ctx context.Context, params *backends.CreateI
 	return new(backends.CreateIndexesResult), nil
 }
 
+// DropIndexes implements backends.Collection interface.
+func (c *collection) DropIndexes(ctx context.Context, params *backends.DropIndexesParams) (*backends.DropIndexesResult, error) {
+	var res backends.DropIndexesResult
+	var err error
+
+	switch {
+	case params.DropAll:
+		err = c.r.IndexesDropAll(ctx, c.dbName, c.name)
+
+	case len(params.Indexes) > 0:
+		var list *backends.ListIndexesResult
+		list, err = c.ListIndexes(ctx, nil)
+
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		for _, index := range params.Indexes {
+			if index == "_id_" {
+				return nil, backends.NewError(backends.ErrorCodeIndexInvalidOptions, nil)
+			}
+
+			if !slices.ContainsFunc(list.Indexes, func(i backends.IndexInfo) bool { return index == i.Name }) {
+				return nil, backends.NewErrorWithArgument(
+					backends.ErrorCodeIndexDoesNotExist,
+					lazyerrors.Errorf("index %q does not exist", index),
+					fmt.Sprintf("index not found with name [%s]", index),
+				)
+			}
+		}
+
+		err = c.r.IndexesDropByNames(ctx, c.dbName, c.name, params.Indexes)
+
+	case len(params.Spec) > 0:
+		// default index {_id: 1} can't be dropped
+		if len(params.Spec) == 1 && params.Spec[0].Field == "_id" && !params.Spec[0].Descending {
```

Comment:
This probably should be handler logic.

Reply:
In principle, this logic could be in the handler, but isn't it a bit too dangerous? :) 

However, if we always define the list of indexes to delete then it should be safe, yeah.

Reply:
Yeah exactly, we define the list to delete :+1: